File: app/schemas/construction.py
import uuid
from datetime import date
from pydantic import BaseModel

class PhaseUpdate(BaseModel):
    phase: str
    status: str
    start_date: date | None = None
    end_date: date | None = None
    notes: str | None = None

class LogCreate(BaseModel):
    phase: str
    log_date: date
    title: str
    content: str | None = None
    photos: list[str] = []

class PaymentCreate(BaseModel):
    phase: str
    payment_type: str
    amount: float
    payee: str | None = None
    payment_date: date
    note: str | None = None
    is_addon: bool = False
    addon_reason: str | None = None

class ChecklistToggle(BaseModel):
    checklist_id: uuid.UUID
    is_checked: bool
    note: str | None = None
    photo_url: str | None = None


class PhaseSeedItem(BaseModel):
    phase: str
    status: str = "pending"
    start_date: date | None = None
    end_date: date | None = None
    notes: str | None = None
    sort_order: int = 0


class ChecklistSeedItem(BaseModel):
    phase: str
    category: str
    content: str
    sort_order: int = 0


class PurchaseSeedItem(BaseModel):
    phase: str
    name: str
    category: str
    estimated_price: float | None = None
    need_measure: bool = False
    note: str | None = None


class PurchaseUpdate(BaseModel):
    purchase_id: uuid.UUID
    is_purchased: bool | None = None
    actual_price: float | None = None
    note: str | None = None


class ConstructionBootstrapRequest(BaseModel):
    phases: list[PhaseSeedItem] = []
    checklists: list[ChecklistSeedItem] = []
    purchases: list[PurchaseSeedItem] = []

File: src/types/construction.ts
export type ConstructionPhase =
    | 'pre_construction'   // 开工前
    | 'demolition'          // 拆改阶段
    | 'hydroelectric'       // 水电阶段
    | 'waterproof'          // 防水阶段
    | 'tiling'              // 瓦工阶段
    | 'carpentry'           // 木工阶段
    | 'painting'            // 油工阶段
    | 'cabinet_install'     // 橱柜安装
    | 'door_window'         // 门窗安装
    | 'final_install'       // 终端安装
    | 'installation'        // 安装阶段 (通用)
    | 'cleaning'            // 保洁阶段
    | 'completed'           // 竣工验收
    | 'warranty';           // 维保期

export interface PhaseInfo {
    phase: ConstructionPhase;
    name: string;
    icon: string;
    order: number;
    typicalDurationDays: number;
    description: string;
}

export interface ConstructionProject {
    id: string;
    houseId: string;
    currentPhase: ConstructionPhase;
    startDate: string;
    estimatedEndDate: string;
    actualEndDate?: string;
    phases: PhaseRecord[];
    logs: ConstructionLog[];
    photos: ConstructionPhoto[];
    payments: PaymentRecord[];
    createdAt: string;
    updatedAt: string;
}

export interface PhaseRecord {
    phase: ConstructionPhase;
    status: 'pending' | 'in_progress' | 'completed' | 'skipped';
    startDate?: string;
    endDate?: string;
    checklistCompleted: number;
    checklistTotal: number;
    notes: string;
}

export interface ConstructionLog {
    id: string;
    date: string;
    phase: ConstructionPhase;
    title: string;
    content: string;
    photos: string[];
    tags: string[];
    createdAt: string;
}

export interface ConstructionPhoto {
    id: string;
    phase: ConstructionPhase;
    url: string;
    thumbnailUrl?: string;
    caption: string;
    category: 'progress' | 'acceptance' | 'problem' | 'material';
    createdAt: string;
}

export interface PaymentRecord {
    id: string;
    phase: ConstructionPhase;
    amount: number;
    description: string;
    payee: string;
    paymentMethod: string;
    receiptUrl?: string;
    date: string;
    createdAt: string;
}

export interface ChecklistItem {
    id: string;
    phase: ConstructionPhase;
    category: string;
    content: string;
    importance: 'critical' | 'important' | 'normal';
    howToCheck: string;
    photoRequired: boolean;
    completed: boolean;
    completedAt?: string;
    note?: string;
}

export interface PurchaseItem {
    id: string;
    phase: ConstructionPhase;
    name: string;
    category: string;
    mustBuyBefore: string;
    needMeasureFirst: boolean;
    estimatedBudget: string;
    estimatedPrice?: number; // 新增：数值型预计单价
    actualPrice?: number;    // 新增：数值型实际单价
    tips: string;
    purchased: boolean;
    purchasedAt?: string;
    actualCost?: number;
}

export interface AppNotification {
    id: string;
    type: 'info' | 'warning' | 'error' | 'success';
    title: string;
    content: string;
    link?: string;
    read: boolean;
    createdAt: string;
}
